### ISSUE TITLE
upgrade to Kissat 4.0.1

### ISSUE THREAD

**Issue by lee-man** [jix/kissat_extras#1]
Hi,

Is there any plan to support incremental solving for kissat 4.0.1?

I am trying to test bitwuzla with kissat 4.0.1, but bitwuzla requires incremental sat solving by default.

Thanks!

**Comment by jix** [comment]
I do plan to release a new version of kissat_extras based on kissat 4.0.1, including full incremental solving support.

Since I didn't keep kissat_extra updated with new kissat releases in the meantime, with a lot of changes accumulating, and also to incorporate some lessons I learned since, I've started over from scratch with kissat 4.0.1. I already made good progress towards adding incremental solving and I think I there is a good chance I will have a first fully working version ready before the end of the year, but I can't quite promise this. Also note that an initial working version will not have seen any optimization and fine tuning for the incremental use case, so it may perform well below what will be possible eventually, in particular it may perform worse than other solvers even for problem domains where non-incremental kissat is usually ahead.
